cypress-io/cypress
# fix: Correctly catch and clear errors during resetErrorAndLoadConfig mutation

- Closes #21785 
- https://github.com/cypress-io/cypress/issues/22389

### User facing changelog


Fixes config re-load issue where the config error message wasn't clearing correctly

### Additional details


The user could not re-load a project configuration after they had clicked the "try again" button while their configuration was still invalid. This was because of two different issues:

- The `resetErrorAndLoadConfig` mutation wasn't catching errors that were thrown in `refreshLifecycle`
- The base error that resulted from the above issue above never got cleared on reset because of a syntax error

See https://github.com/cypress-io/cypress/issues/22389 for more info

### Steps to test


I tested this by introducing a configuration error in my project config and then opening the project in Cypress. You'll be presented with an error page that has the "Try Again" button. 

- Click the try again button while the config is still invalid
- Verify that the error title does not disappear from the page
- Fix the configuration
- Click the try again button again
- Verify that the error clears and you are taken to the "Welcome to Cypress" screen

### PR Tasks


- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
